packages/data-explorer/src/components/DataRowEdit.vue

```diff
@@ -0,0 +1,266 @@
+        inputDebounceTime: 500
+      },
+      showRef: false,
+      idAttribute: null,
+      labelAttribute: null,
+      referenceMap: {} // Map from field id to entityName for all reference entities
+    }
+  },
+  methods: {
+    onValueChanged (updatedFormData) {
+      this.formData = updatedFormData
+    },
+    onAddOptionRequest (optionCreatedCallback, event, sourceField) {
+      // Translate the reference field id (short name) into a tableId (full molgenis name)
+      const referenceTableId = this.referenceMap[sourceField.id]
+
+      // Store the afterOption creation callback on the parent
+      this.optionCreatedCallback = optionCreatedCallback
+
+      // Create a new child DataRowEdit for the reference option passing the parent for context
+      const ComponentClass = Vue.extend(DataRowEdit)
+      const refDataRowEdit = new ComponentClass({
+        propsData: { entity: referenceTableId, parent: this }
+      })
+      refDataRowEdit.$mount()
+
+      // Show the reference option form and hide the parent form
+      this.setRef(true) // hide parent
+      this.$refs.refContainer.appendChild(refDataRowEdit.$el) // show child
+    },
+    touchField (field) {
+      field.children ? field.children.forEach(this.touchField) : this.formState[field.id].$touched = true
+    },
+    async onSubmit () {
+      this.formFields.forEach(this.touchField) // validate all fields
+
+      if (this.formState.$valid) {
+        this.isSaving = true
+        try {
+          const response = await repository.save(this.formData, this.formFields, this.entity, this.dataRowId)
+          if (this.parent) {
+            const newOptionLocation = response.headers.get('Location')
+            const createdRow = await repository.fetchOption(newOptionLocation)
+            // Create a new option object to pass to the reference select
+            this.parent.optionCreatedCallback({
+              id: createdRow[this.idAttribute],
+              value: createdRow[this.idAttribute],
+              label: createdRow[this.labelAttribute]
+            })
+            this.showParent()
+          } else {
+            this.goBackToPluginCaller()
+          }
+        } catch (e) {
+          this.handleError(e)
+        }
+      } else {
+        this.saveFailed = true
+      }
+    },
+    onCancelClick () {
+      this.parent ? this.showParent() : this.goBackToPluginCaller()
+    },
+    goBackToPluginCaller () {
+      window.history.go(-1)
+    },
+    clearAlert () {
+      this.alert = null
+    },
+    setRef (ref) {
+      this.showRef = ref
+    },
+    showParent () {
+      this.parent.setRef(false) // show parent
+      this.parent.$refs.refContainer.removeChild(this.$el) // destroy child
+    },
+    handleError (error) {
+      const alertMsg = this.errorToMessage(error)
+      this.showForm = true
+      this.isSaving = false
+      this.alert = { message: alertMsg, type: 'danger' }
+    },
+    errorToMessage (error) {
```

Comment:
should probably be replaced by more generic data explorer error handler

Reply:
Refactoring the DataRowEdit component is out of scope for this ticket.